### ISSUE TITLE
Add workarounds for ClickOnce (ResolveAssemblyReferences) related issues 

### ISSUE THREAD

**Issue by Daniel-Svensson** [onovotny/MSBuildSdkExtras#58]
Would you be interested in adding workarounds for issues with publishing (clickonce) in the build process which are caused by how sdk style projects and PackageReference incorrectly sets "Private=True" for references?

Issues:
* Netstandard support - breaks clickonce (ResolveAssemblyReferences problem)
See https://github.com/Microsoft/msbuild/issues/3057
* Using PackageReference and ClickOnce in the same Project (ResolveAssemblyReferences problem) does not work
https://github.com/Microsoft/msbuild/issues/3058 )

Workaround for both of these issues can be done in  the project file by adding the following code at the end so I hope it should be easy to add to the extras sdk, but I am very unsure about where to put it so that is why I put it here and dont post a  PR.

The following code manually create "PublishFile" items for items which would otherwise be classified incorrectly.

```
<Target Name="EnsureNetstandardFilesAreIncludedInClickOnce" BeforeTargets="_DeploymentComputeClickOnceManifestInfo">
    <Message Text="Adding netstandard files to ClickOnce" />
    <ItemGroup>
      <PublishFile Include="@(_NETStandardLibraryNETFrameworkLib->'%(FileName)')" KeepMetadata="None">
        <PublishState>Include</PublishState>
        <FileType>Assembly</FileType>
      </PublishFile>
    </ItemGroup>
    <!-- Consider using ReferenceCopyLocalPaths instead, if we are missing files -->
    <Message Text="Adding nuget files to ClickOnce" />
    <ItemGroup>
      <PublishFile Include="@(ReferencePath->'%(FileName)')" Condition=" '%(ReferencePath.NuGetSourceType)' == 'Package' and '%(ReferencePath.NuGetIsFrameworkReference)' != 'true'" KeepMetadata="None">
        <PublishState>Include</PublishState>
        <FileType>Assembly</FileType>
      </PublishFile>
    </ItemGroup>

    <Message Text="Adding Private True references to ClickOnce" />
    <ItemGroup>
      <PublishFile Include="@(ReferencePath->'%(FileName)')" Condition="'%(ReferencePath.Private)' == 'True'" KeepMetadata="None">
        <PublishState>Include</PublishState>
        <FileType>Assembly</FileType>
      </PublishFile>
    </ItemGroup>
	</Target>
```

**Comment by Daniel-Svensson** [comment]
No i haven't tried that version.
Still running last 15.6 build. 
But I just went through all issues closed in the msbuild repo for the last month and none if them sounded like it would/could fix any of the issues

**Comment by Daniel-Svensson** [comment]
The same issue exist for 15.7 , but in addition `Exclude="netfx.force.conflicts"` must be added to the first netstandard workaround

**Comment by techcap** [comment]
Still doesn't work for 15.7.5.

**Comment by onovotny** [comment]
I saw a report on the main ClickOnce issue thread that it appears to be fixed in 15.8 preview 5.

**Comment by techcap** [comment]
When I used this workaround, compile error is solved. But nuget ref dll instead of lib dll is published. In project build output, lib dll is generated. 

Testing procedure
1. Create wpf application
2. Add nuget reference of System.Buffers v4.5.
3. ClickOnce publish. It generates proper lib dll of System.Buffers.dll.deploy.
4. Migrate packages.config to PakcageReference
5. Apply this workaround
6. ClickOnce publish, It generates wrong ref dll of System.Buffers.dll.deploy.

**Comment by Daniel-Svensson** [comment]
@onovotny @techcap the issues seems to be resolved in 15.8

Be ware of issues in 15.8.1 if any project include localized Resources https://developercommunity.visualstudio.com/content/problem/317922/wrong-path-to-resource-in-manifest-for-clickonce-a.html
